samples/CppArExample/CppArExample.cpp

```diff
@@ -0,0 +1,353 @@
+// Copyright 2019 ESRI
+//
+// All rights reserved under the copyright laws of the United States
+// and applicable international laws, treaties, and conventions.
+//
+// You may freely redistribute and use this sample code, with or
+// without modification, provided you include the original copyright
+// notice and use restrictions.
+//
+// See the Sample code usage restrictions document for further information.
+
+#include "CppArExample.h"
+
+#include "ArcGISTiledElevationSource.h"
+#include "Basemap.h"
+#include "Scene.h"
+#include "SceneQuickView.h"
+#include "IntegratedMeshLayer.h"
+#include "ArcGISSceneLayer.h"
+#include "ArcGISArView.h"
+#include "GeometryEngine.h"
+#include "PointCloudLayer.h"
+#include "PortalItem.h"
+#include "IntegratedMeshLayer.h"
+#include "LocationDataSource.h"
+
+#include <QUrl>
+#include <QFileInfo>
+#include <QStandardPaths>
+
+#include <QGeoPositionInfoSource>
+
+#include "SimpleMarkerSceneSymbol.h"
+#include "SimpleLineSymbol.h"
+
+using namespace Esri::ArcGISRuntime;
+using namespace Esri::ArcGISRuntime::Toolkit;
+
+CppArExample::CppArExample(QObject* parent):
+  QObject(parent)
+{
+}
+
+CppArExample::~CppArExample()
+{
+}
+
+ArcGISArView* CppArExample::arcGISArView() const
+{
+  return m_arcGISArView;
+}
+
+void CppArExample::setArcGISArView(ArcGISArView* arcGISArView)
+{
+  if (!arcGISArView || arcGISArView == m_arcGISArView)
+    return;
+
+  m_arcGISArView = arcGISArView;
+
+  emit arcGISArViewChanged();
+}
+
+SceneQuickView* CppArExample::sceneView() const
+{
+  return m_sceneView;
+}
+
+// Set the view (created in QML)
+void CppArExample::setSceneView(SceneQuickView* sceneView)
+{
+  if (!sceneView || sceneView == m_sceneView)
+    return;
+
+  m_sceneView = sceneView;
+  emit sceneViewChanged();
+}
+
+// properties for debug mode
```

Comment:
@GuillaumeBelz - Should this be?

`// Sets whether point clouds should be visible.`